docs/clusterjoinprocess.md

```diff
@@ -17,28 +17,26 @@ and deploy the agent on managed cluster.
   - it has the identity to create `ManagedCluster` and create/watch csr.
 2. agent on managed cluster creates `ManagedCluster` if it does not exist.
   - The name of `ManagedCluster` is read from Cluster UID in openshift.
-  - Otherwise agent generates a UID and saves it in `Configmap` in managed cluster, so restarting agent or redeploying
-  agent will not lose the UID.
+  - Otherwise agent generates an agent UID and use it as part of `CommonName` to request a certificate(step 3), and once the CSR is approved on the hub, the agent on the managed cluster will store it as `agent-name` in `Secret` `hub-kubeconfig-secret`(step 10), so restarting agent or redeploying agent will not lose the UID after the cluster is managed successfully.
 3. agent on managed cluster creates CSR on hub cluster using bootstrap kubeconfig.
-  - The subject in csr is
+  - The subject in CSR is
 `{"Organization": ["system:open-cluster-management:clusterName"], "CommonName":"system:open-cluster-management:clusterName:agentName"}`.
   - The name of the csr is the digest of subject and private key, with a common prefix.
   CSR will specify the signer name as the kube-client one.
 4. cluster-admin on hub-cluster approve the CSR.
 5. hub-controller creates a clusterrolebinding on the hub with the identity of
-`system:open-cluster-management:clusterName:agentName`
+`open-cluster-management:managedcluster:clusterName`
    - Allows status update of `ManagedCluster`
 6. cluster-admin on hub update `spec.hubAcceptsClient` to `true`.
-  - Only user on hub who has the RBAC permision to update subresource of `managedclusters/accept`
+  - Only user on hub who has the RBAC permission to update subresource of `managedclusters/accept`
   can update this field.
-7. hub-controller updates condition of `ManagedCluster` to `HubApprovedJoin`.
+7. hub-controller updates condition of `ManagedCluster` to `HubAcceptedManagedCluster`.
 8. hub-controller creates a namespace as the name of cluster on hub cluster if it does not exist.
   - managed cluster can only join a hub once, and it can join to multiple hubs.
   - The UID of the managed cluster is identical on each of the hub the Klusterlet agent joins.
```

Comment:
Here I have a question: as far as I know, cluster join starting with a bootstrap-secret(specified by a flag of the spoken agent, for example, `bootstrap-secret`),  there only can be **one** `bootstrap-secret`(representing one hub) in the agent running namespace, so I do not know how to register one managed cluster to multiple hubs by one agent? Of course, we can run multiple agents in separate namespaces to register the cluster to multiple hubs, but I am curious how to keep the UID identical.

Reply:
UID here is cluster name, and it is identical in hub 1, it should also be identical in hub2.

Reply:
Sorry I got it wrong, thanks for the explanation